### PR TITLE
problems: add camisole debug i/o to traces

### PR DESCRIPTION
**Pull request by seirl**

### CODE REVIEW

**Comment by zopieux** [comment]
The thing is, what will you do when we finally use webpack instead of JSON? I guess we're better of using `pformat` instead of `json.dumps`, as the message serialization is not relevant and will be obsolete when camisole 1.0 is merged.

File: prologin/problems/models/submission.py [hunk 1]
```diff
@@ -190,6 +193,10 @@ def get_score_func(prefix=None):
     ScoreFunc = get_score_func()
 
 
+def _strip_str(s, l=100):
+    return s[:l] + '...' if len(s) > l else s
```
Comment (zopieux) [inline prologin/problems/models/submission.py]: `…`

File: prologin/problems/models/submission.py [hunk 2]
```diff
@@ -232,6 +240,20 @@ def correction_results(self):
             return None
         return Result.parse(self.submission.problem_model(), self.result)
 
+    def request_printable(self):
+        req = copy.deepcopy(self.request)
+        req['source'] = _strip_str(req['source'])
+        for test in req['tests']:
+            test['stdin'] = _strip_str(test['stdin'])
+        return json.dumps(req, sort_keys=True, indent=4)
```
Comment (zopieux) [inline prologin/problems/models/submission.py]: Use `pformat`

File: prologin/problems/models/submission.py [hunk 3]
```diff
@@ -232,6 +240,20 @@ def correction_results(self):
...
+
+    def result_printable(self):
+        rep = copy.deepcopy(self.result)
+        for test in rep['tests']:
+            test['stdout'] = _strip_str(test['stdout'])
+            test['stderr'] = _strip_str(test['stderr'])
+        return json.dumps(rep, sort_keys=True, indent=4)
```
Comment (zopieux) [inline prologin/problems/models/submission.py]: Use `pformat`

**Review by zopieux — CHANGES_REQUESTED** [review]

File: prologin/problems/models/submission.py [hunk 4]
```diff
@@ -232,13 +240,53 @@ def correction_results(self):
...
         return reverse('problems:submission', kwargs={
-            'year': problem.challenge.year, 'type': problem.challenge.event_type.name,
+            'year': problem.challenge.year,
+            'type': problem.challenge.event_type.name,
             'problem': problem.name, 'submission': self.id,
         })
 
+    def generate_request(self) -> dict:
+        """
+        Generate a Camisole request for the given SubmissionCode.
```
Comment (zopieux) [inline prologin/problems/models/submission.py]: **c**amisole

File: prologin/problems/models/submission.py [hunk 5]
```diff
@@ -190,6 +193,11 @@ def get_score_func(prefix=None):
     ScoreFunc = get_score_func()
 
 
+def _strip_str(s, l=100):
+    append = b'...' if isinstance(s, bytes) else '...'
```
Comment (zopieux) [inline prologin/problems/models/submission.py:196]: what about `…`

Comment (seirl) [reply prologin/problems/models/submission.py:196]: Ugly with bytes

**Review by zopieux — APPROVED** [review]

**Pull request merged**